packages/autoloader/src/AutoloadGenerator.php

```diff
@@ -0,0 +1,311 @@
+<?php
+// phpcs:ignoreFile -- this is not a core file
+/**
+ * Autoloader Generator.
+ */
+namespace Jetpack\Autoloader;
+
+use Composer\Autoload\AutoloadGenerator as BaseGenerator;
+use Composer\Autoload\ClassMapGenerator;
+use Composer\Config;
+use Composer\Installer\InstallationManager;
+use Composer\IO\IOInterface;
+use Composer\Package\PackageInterface;
+use Composer\Repository\InstalledRepositoryInterface;
+use Composer\Util\Filesystem;
+
+/**
+ * Class AutoloadGenerator.
+ */
+class AutoloadGenerator extends BaseGenerator {
+
+	/**
+	 * @var IOInterface
+	 */
+	private $io;
+
+	/**
+	 * @var bool
+	 */
+	private $classMapAuthoritative = false;
+
+	/**
+	 * Instantiate a AutoloadGenerator object.
+	 *
+	 * @param IOInterface $io
+	 */
+	public function __construct( $io ) {
+		$this->io = $io;
+	}
+
+	/**
+	 * Whether or not generated autoloader considers the class map
+	 * authoritative.
+	 *
+	 * @param bool $classMapAuthoritative
+	 */
+	public function setClassMapAuthoritative( $classMapAuthoritative ) {
```

Comment:
could we remove this method? I think it doesn't difer from the parent

Reply:
removed